tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -777,6 +778,49 @@ public function testConnectionParamsArePassedToTheQueryCacheProfileInExecuteCach
         );
     }
 
+    public function testShouldNotPassPlatformInParamsToTheQueryCacheProfileInExecuteCacheQuery()
+    {
+        $resultCacheDriverMock = $this->createMock(Cache::class);
+
+        $resultCacheDriverMock
+            ->expects($this->atLeastOnce())
+            ->method('fetch')
+            ->with('cacheKey')
+            ->will($this->returnValue(['realKey' => []]));
+
+        $query  = 'SELECT * FROM foo WHERE bar = ?';
+        $params = [666];
+        $types  = [\PDO::PARAM_INT];
+
+        /* @var $queryCacheProfileMock QueryCacheProfile|\PHPUnit_Framework_MockObject_MockObject */
+        $queryCacheProfileMock = $this->createMock(QueryCacheProfile::class);
+
+        $queryCacheProfileMock
+            ->expects($this->any())
+            ->method('getResultCacheDriver')
+            ->will($this->returnValue($resultCacheDriverMock));
+
+        $connectionParams = $this->params;
+        $expectedParams = $connectionParams;
+
+        $connectionParams["platform"] = $this->createMock(AbstractPlatform::class);
+
+        // This is our main expectation
+        $queryCacheProfileMock
+            ->expects($this->once())
+            ->method('generateCacheKeys')
+            ->with($query, $params, $types, $expectedParams)
+            ->will($this->returnValue(['cacheKey', 'realKey']));
+
+        /* @var $driver Driver */
+        $driver = $this->createMock(Driver::class);
+
+        $this->assertInstanceOf(
+            ArrayStatement::class,
+            (new Connection($connectionParams, $driver))->executeCacheQuery($query, $params, $types, $queryCacheProfileMock)
```

Comment:
Removing the assert as it is tested by "testConnectionParamsArePassedToTheQueryCacheProfileInExecuteCacheQuery"